lib/ExpensiMark.js

```diff
@@ -114,6 +116,19 @@ export default class ExpensiMark {
                 replacement: '<h1>$1</h1>',
             },
 
+            /**
+             * Converts markdown style images to img tags e.g. ![Expensify](https://www.expensify.com/attachment.png)
+             * We need to convert before linking rules since they will not try to create a link from an existing img
+             * tag.
+             * Additional sanitization is done to the alt attribute to prevent parsing it further to html by later rules.
+             */
+            {
+                name: 'image',
+                regex: MARKDOWN_IMAGE_REGEX,
+                replacement: (match, g1, g2) => `<img src="${Str.sanitizeURL(g2)}" alt="${this.escapeMarkdownEntities(g1)}" />`,
+                rawInputReplacement: (match, g1, g2) => `<img src="${Str.sanitizeURL(g2)}" alt="${this.escapeMarkdownEntities(g1)}" data-raw-href="${g2}" data-link-variant="labeled" />`
```

Comment:
Escaping is good - but why display them at all? The alt tag will show a bunch of junk if the image fails to load or they hover over it.

Reply:
I considered keeping the text plain initially, but realized that it wouldn't allow us to revert to the original Markdown message from the HTML. This way, the original format is retained.

Regarding hover text, the `alt` attribute doesn't display text on hover. For that, Markdown has an extended syntax (`![alt](src "title text")`) that adds a `title` attribute to the resulting HTML. However, we don't support this feature, similarly to our approach with links.

If an image fails to load, the fallback is something like:
![image](https://github.com/Expensify/expensify-common/assets/12156624/6647842a-9fbc-404c-8dba-1908c54d2d52)

But NewDot and react-native-web don't display this fallback since they render a hidden `<img>` tag instead.

Reply:
Ah hmm. Quite weird. Perhaps we can filter it before rendering i.e. not actually modify the underlying `html` field. But we can look into that in the App PR. Thanks for the explanation.